code/modules/food_and_drinks/food/candy.dm

```diff
@@ -134,14 +128,15 @@
 	trash = /obj/item/trash/candy
 	filling_color = "#7D5F46"
 	bitesize = 3
-	list_reagents = list("nutriment" = 2, "chocolate" = 5)
+	junkiness = 25
```

Comment:
The intention is to discourage vending machine use and encourage going to the chef--it's a bit like shooting yourself in the foot to have chef produced food do that; but yes, thanks for pointing that out; I'll have to correct that as it wasn't intentional.